cmd/snap/cmd_run.go

```diff
@@ -1509,6 +1536,25 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 				return err
 			}
 		}
+
+		// If a journal namespace is supplied for the service, then we reopen stdout/stderr
+		// connected to that journal namespace instead of the main journal. Since we are not
+		// using systemd's LogNamespace= directly, we must do this ourselves.
+		if lns := os.Getenv("SNAPD_LOG_NAMESPACE"); lns != "" {
+			stdout, stderr := makeStdStreamsForJournal(app, lns)
+			if stdout != nil {
+				defer stdout.Close()
+				if err := sys.Dup2(stdout.Fd(), uintptr(syscall.Stdout)); err != nil {
```

Comment:
could we always use syscall.Dup3? maybe wrapping it in osutil.DupFileFD() or something similar

Reply:
While it probably doesn't matter, the reasoning was that dup2 is actually a part of the POSIX standard, and dup3 is linux-specific. It felt more safe to stick to dup2.

Reply:
I think it is confusing that Dup2 actually calls Dup3 on linux. So maybe wrap it like @bboozzoo said. So it does not introduce confusion.

Reply:
Changed, thanks for the input!